api/mapping/models.py

```diff
@@ -499,32 +510,31 @@ class StructuralMappingRule(BaseModel):
     """
     To come
     """
-
+    #save the scan_report link to make it easier
     scan_report = models.ForeignKey(ScanReport, on_delete=models.CASCADE)
 
     omop_field = models.ForeignKey(OmopField, on_delete=models.CASCADE)
 
-    source_table = models.ForeignKey(
-        ScanReportTable, on_delete=models.CASCADE, blank=True, null=True
-    )
+    source_field = models.ForeignKey(ScanReportField,
+                                     on_delete=models.CASCADE,
+                                     null=True,
+                                     blank=True)
 
-    source_field = models.ForeignKey(
-        ScanReportField,
-        on_delete=models.CASCADE,
-        blank=True,
-        null=True
-        # limit_choices_to= {'scan_report_table': source_table}
-    )
+    #term_mapping = models.JSONField(null=True,blank=True)
+    do_term_mapping = models.BooleanField(default=False)
```

Comment:
What does do_term_mapping do?

Reply:
adding the following comment
```
# Flag if this is a rule that should be using term_mapping or not                                                  
    # - rules such as source_value should apply no term mapping                                                        
    # - we need a flag as we want to preserve a link to the ScanReportConcept                                          
    # - this is because, if that is deleted, we want to remove the rule                                                
    # - however, we dont want to ever apply any of the concept_id term_mappings                                        
    #   if this was source_value, person_id, birth_date etc.                                                           
    do_term_mapping = models.BooleanField(default=False)
```